githubwebhooks/iam-roles.tf

```diff
@@ -146,3 +146,51 @@ resource "aws_iam_role_policy" "lambda_github_webhooks_pulse" {
     role = "${aws_iam_role.lambda_github_webhooks_pulse.id}"
     policy = "${data.aws_iam_policy_document.lambda_github_webhooks_pulse.json}"
 }
+
+data "aws_iam_policy_document" "sns_webhooks_all" {
+    # Give self full access.
+    statement = {
```

Comment:
I'm not too familiar with TerraForm but if this is you granting your own account rights via a resource policy (AKA SNS Topic Policy), I'd recommend against it unless there's a good reason. I'd recommend governing access to your SNS topics in your own account with an [identity-based policy](http://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html) (aka an IAM policy).

Reply:
When I loaded up the AWS web console, the topic had a default policy which contained the rules I implemented here. Not sure why that is or if anything is relying on it.

We already have a separate IAM policy allowing the Lambda function to SNS:Publish to this topic. That's the only thing that matters. So I'll remove this.

If we lock ourselves out of the topic, we can restore access via an IAM policy elsewhere.